### PR TITLE
Add api login to apple auth provider

### PR DESCRIPTION
**Pull request by 0aps**
Use anxeb auth service for apple social login.

**I'm not a fan of this approach but is the solution that requires less changes on the current API.**

### CODE REVIEW

File: lib/parts/auths/apple.dart [hunk 1]
```diff
@@ -5,10 +5,12 @@ import 'package:sign_in_with_apple/sign_in_with_apple.dart';
 class AppleAuth extends AuthProvider {
   String _fetchCallbackRoute;
   String Function() _nonce;
+  Api _api;
 
   AppleAuth(Application application) : super(application) {
     _fetchCallbackRoute = application.settings.auths.apple.fetchCallbackRoute;
     _nonce = application.settings.auths.apple.nonce;
+    _api = application.apiLogin ?? application.api;
```
Comment (ingerosario) [inline lib/parts/auths/apple.dart]: Got it, but instead creating a new property inside general application class, you can use the private _AuthApple class (middleware/settings.dart) for this matter.

Take the line above as an example, private variable _nonce is setted from that class.

File: lib/parts/auths/apple.dart [hunk 2]
```diff
@@ -26,7 +28,7 @@ class AppleAuth extends AuthProvider {
       );
 
       if (session != null) {
-        var res = await application.api.post(_fetchCallbackRoute, {
+        var res = await _api.post(_fetchCallbackRoute, {
```
Comment (ingerosario) [inline lib/parts/auths/apple.dart:31]: This is ok as long as _api comes from _AuthApple settings class. Remember to still use the same application.api if _api is null

**Review by ingerosario — CHANGES_REQUESTED** [review]
Please check incode comments

File: lib/parts/auths/apple.dart (hunk 1 above)
Comment (0aps) [reply lib/parts/auths/apple.dart]: ready

File: lib/parts/auths/apple.dart (hunk 2 above)
Comment (0aps) [reply lib/parts/auths/apple.dart:31]: ready

**Comment by ingerosario** [comment]
there you go!

**Pull request merged**